bundles/tools.mdsd.ecoreworkflow.mwe2lib/src/tools/mdsd/ecoreworkflow/mwe2lib/component/RegexComponent.java

```diff
@@ -32,75 +34,165 @@
+            stream.write(content.getBytes(charset));
+        }
+    }
+
+    protected List<URI> determineFilesToReplace(Replacement replacement) throws IOException {
+        Optional<URI> directory = Optional.ofNullable(replacement.getDirectory())
+            .map(this::fromFilename)
+            .map(this::ensureAbsoluteURI)
+            .map(this::ensureEmptyLastSegment);
+
+        List<URI> filesToProcess = replacement.getFilenames()
+            .stream()
+            .map(this::fromFilename)
+            .map(uri -> directory.map(uri::resolve)
+                .orElse(uri))
+            .collect(Collectors.toList());
+
+        if (replacement.getWildcard() != null && directory.isPresent()) {
+            final PathMatcher matcher = FileSystems.getDefault()
+                .getPathMatcher("glob:" + replacement.getWildcard());
+
+            var dirPath = getAbsolutePathOfDirectory(directory.get());
+
+            Files.walkFileTree(dirPath, new SimpleFileVisitor<Path>() {
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                    var relativeMatch = dirPath.relativize(file);
+                    if (matcher.matches(relativeMatch)) {
+                        filesToProcess.add(URI.createURI(file.toUri()
+                            .toString()));
+                    }
+                    return FileVisitResult.CONTINUE;
+                }
+
+                @Override
+                public FileVisitResult visitFileFailed(Path file, IOException exc) throws IOException {
+                    return FileVisitResult.CONTINUE;
+                }
+            });
+        }
+        return filesToProcess;
+    }
+
+    protected URI fromFilename(String filename) {
+        var uri = parseURI(filename);
+        return uri.orElseGet(() -> URI.createURI(Path.of(filename)
+            .toUri()
+            .toString()));
+    }
+
+    protected Optional<URI> parseURI(String uri) {
+        try {
+            var jUri = new java.net.URI(uri);
+            return Optional.of(URI.createURI(jUri.toString()));
+        } catch (URISyntaxException e) {
+            return Optional.empty();
+        }
+
+    }
+
+    protected Path getAbsolutePathOfDirectory(URI uri) {
+        Path result;
+        if (uri.isPlatform()) {
+            var platformString = uri.toPlatformString(true);
+            if (Platform.isRunning()) {
+                var path = new org.eclipse.core.runtime.Path(platformString);
+                result = ResourcesPlugin.getWorkspace()
+                    .getRoot()
+                    .getFile(path)
+                    .getLocation()
+                    .toFile()
+                    .toPath();
+            } else {
+                var resolvedUri = EcorePlugin.resolvePlatformResourcePath(platformString);
+                result = getPathOfFileURI(resolvedUri);
+            }

```

Comment:
This code does not seem to be used. At least, all calls to this method I found submit an absolute URI. Is there an intended future use? If so, it would be nice to include this branch in the unit test.